### ISSUE TITLE
[beta2] some small issues

### ISSUE THREAD

**Issue by gnzgg** [hydrogen-music/hydrogen#826]
Hydrogen version *          : 1.0.0 beta2
Operating system + version  : Debian (Buster)
Audio driver + version      :  Jack1 / Jack2 / Alsa

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Hey, so I've been trying to set up hydrogen as a kind of multi-function drum sequencer, that would use both sampled percussions and midi outputs to synths using a small midi controller to work with it 

There are many little things I'm not able to get right just yet, some of them are very likely related to me being a NOOB, sorry if that's the case :) 
Still, I figured I might as well make a list and see if some of y'all can help me with some of it, so here we go !

1: [WORKAROUNDED FOR NOW] MIDI inconsistencies:
whenever I try to plug Hydrogen's MIDI OUT to another instrument, it's pretty wildly inconsistent, with the notes on/notes off messages kinda stepping on one another in a pattern like this 
ON / OFF / ON / OFF / OFF / OFF / ON / OFF / OFF / ON

Now maybe it's just my setup, like an issue with my audio drivers or something, I genuinely don't know, but I have tried with different audio drivers, sample rates and midi drivers (seq, raw, jack) and it doesn't change anything so I'm kinda stumped here

2: MIDI Strip Mute: 
Muting an instrument doesn't mute its MIDI output
This has already been discussed on this board and I can't seem to find the reason why it's not working for me, has it not been implemented yet or am I just doing something wrong ?
One workaround for me would be to rebuild it using the patch that's in the discussion I mentioned, but considering how important a feature this is, I would rather address it here

EDIT: Rereading this, the rest of the list is more in the feature request territory, maybe I should make a separate thread for them with a better tagline ?

3: MIDI MAP Strip specific control: 
I would love to be able to map a midi CC knob to the strip that's currently selected, so that would include A/D/S/R, gain and filter cutoff/resonance, and then whenever I select another strip the CC switches accordingly

4: [Solved by #525 ] MIDI MAP Humanize on the master bus
The Humanize Velocity / Timing / Swing function can't be mapped to MIDI yet it seems, that would be a nice addition 

5: Instrument specific note on/off options 
So there's a "ignore note-off" option in the MIDI section of the Preferences which is pretty useful, but I think an Instrument specific option would also be really important, there are many cases where you might want some instruments to receive note off signals and other to not receive them

Anyway, that's all I can think of right now, I would really appreciate if you can help me solve some of them, 
I am otherwise very satisfied with this software so far, it really seems to have everything I've been looking for to have a viable Linux/FLOSS setup for live music and I'm very glad to see this project being active !

**Comment by elpescado** [comment]
Hi!

> 1: MIDI inconsistencies:
> whenever I try to plug Hydrogen's MIDI OUT to another instrument, it's pretty wildly inconsistent, with the notes on/notes off messages kinda stepping on one another in a pattern like this
> ON / OFF / ON / OFF / OFF / OFF / ON / OFF / OFF / ON

Note Off handling in Hydrogen is kinda weird (and in my opinion, broken). It's tied to sample length, so Note Off message is sent when sample playback stops. It does not make much sense, for example one consequence is that you have to have some samples, even if you want pure MIDI setup. There has been discussion about this but no conslusions were reached.

> 4: MIDI MAP Humanize on the master bus
> The Humanize Velocity / Timing / Swing function can't be mapped to MIDI yet it seems, that would be a nice addition

I've implemented that in #525, along with some more goodies;)

**Comment by gnzgg** [comment]
Hi, thanks for the quick answer, so 1: can be worked around decently if I use really short samples as my midi triggers and set my inputs accordingly
I can understand why solving this may prove to be a struggle, hoping that will eventually get fixed, because that would really take things to the next level

as for 4: I'm not very familiar with branching on github and all of that so I guess I'm gonna RTFM and get into it, it seems like a nice addition

**Comment by elpescado** [comment]
There's another workaround. If you want exact note duration, you could right-click on note in pattern editor and right-drag it setting expected duration. The note changes itself from "circle" to "rectangle".

**Comment by thijz** [comment]
hi @gnzgg 

can you please have a look at this wiki page ?
https://github.com/hydrogen-music/hydrogen/wiki/MIDI-implementation---current-state-and-usecases

it contains an (incomplete) listing of all things wrong with H2 midi
please feel free to add your insights (assuming you have edit rights .... not sure about that )

**Comment by gnzgg** [comment]
@elpescado Indeed !
one quirk is that you need the sample to be longer than the midi note, otherwise, the end of the sample will send the NOTE OFF signal and cuts the note early
tbh I think the fix would simply be to completely remove the samplelength-to-midi component (or put it as an option burried in the settings i guess) and send midi signals strictly based on note length
Of course, for "circle" notes that implies a very short "default" duration that maybe you could tweak in the settings somewhere (they would actually still be useful as very simple midi triggers)

@thijz Looking into it, the page seems to allow for public edits, I'm probably going to compile and verify all the issues I can find and try to document them as much as possible and add my grain of salt to it, thanks for the suggestion :)

**Comment by thijz** [comment]
> one quirk is that you need the sample to be longer than the midi note, otherwise, the end of the sample will send the NOTE OFF signal and cut the note early

checj out this ticket : https://github.com/hydrogen-music/hydrogen/issues/63

**Comment by trebmuh** [comment]
@gnzgg : could you please split off all those thing in different issues on github please. It will allow for very much **more easily** management of those on a long term view since thread with a lot of thing tends to grow up a lot, mixing a lot of different things, and then die after a while because no-one want to re-read a whole thread with 80% OT when just interested by 1 thing in it.

TL;DR: one question = one thread

:sunglasses:

**Comment by thijz** [comment]
@trebmuh : indeed
@gnzgg : also check out all the 'MIDI related' tickets.  you will find a lot of your remarks are already logged (but still need fixing :-)

**Comment by gnzgg** [comment]
Thanks for your feedback, I'm closing this thread and will pay more attention to not leaving a mess on this board :)
